config: rebalancer sharding role (#9486)
tarantool/tarantool: changelogs/unreleased/gh-8862-sharding-role-rebalancer.md

## feature/config

> * The sharding.role option has been disabled in the instance scope (gh-8862).
> * The 'rebalancer' sharding role was instroduced (gh-8862).

Reviewer: ```suggestion
* Disabled the `sharding.role` option in the instance scope (gh-8862).
* Introduced the `rebalancer` sharding role (gh-8862).
```

Author: Thanks! Fixed. The day will come when I will write a changelog without errors. I hope ... :smile:
